modules/web/src/app/settings/admin/presets/dialog/steps/settings/provider/vsphere/component.ts

```diff
@@ -78,11 +80,16 @@ export class VSphereSettingsComponent extends BaseFormValidator implements OnIni
     delete this._presetDialogService.preset.spec.vsphere;
   }
 
+  onNetworksChange(networks: string[]): void {
+    this.networks = networks;
+    this.form.get(Controls.Networks).updateValueAndValidity();
+  }
+
   private _update(): void {
     this._presetDialogService.preset.spec.vsphere = {
       username: this.form.get(Controls.Username).value,
       password: this.form.get(Controls.Password).value,
-      vmNetName: this.form.get(Controls.VMNetName).value,
+      networks: this.form.get(Controls.Networks).value,
```

Comment:
```suggestion
      networks: this.form.get(Controls.Networks).value?.tags,
```

Reply:
Do we really need `tags` here?

Reply:
Yes, because `km-chip-list` has internal form which has `tags` form control so in order to get the actual value we need to do this. Otherwise, we get `tags` object in request payload.

Reply:
> networks: this.form.get(Controls.Networks).value?.tags,

Yes, we do Waleed 🤦